include/cuco/static_multimap.cuh

```diff
@@ -0,0 +1,1716 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cooperative_groups.h>
+#include <thrust/distance.h>
+#include <thrust/functional.h>
+#include <cub/cub.cuh>
+#include <memory>
+
+#include <cuco/allocator.hpp>
+
+#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11000) && defined(__CUDA_ARCH__) && \
+  (__CUDA_ARCH__ >= 700)
+#define CUCO_HAS_CUDA_BARRIER
+#endif
+
+// cg::memcpy_aysnc is supported for CUDA 11.1 and up
+#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11100)
+#define SUPPORTS_CG_MEMCPY_ASYNC 1
+#else
+#define SUPPORTS_CG_MEMCPY_ASYNC 0
+#endif
```

Comment:
This would be the more canonical way to do this. 
```suggestion
// cg::memcpy_aysnc is supported for CUDA 11.1 and up
#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11100)
#define CUCO_HAS_CG_MEMCPY_ASYNC
#endif
```